.github/workflows/test-build-deploy.yml

```diff
@@ -278,91 +281,120 @@ jobs:
       - name: Build Mimir with race-detector
         run: |
           export IMAGE_TAG_RACE=$(make image-tag-race)
-          export MIMIR_IMAGE="grafana/mimir:$IMAGE_TAG_RACE"
+          # We are working on migrating to a distroless image as our only base.
+          # For now we mark our distroless image with a tag prefix `distroless-`
+          # and Alpine remains the default image pushed to the registry.
+          # We always test the distroless image, and we test the alpine image on deploy.
+          export MIMIR_IMAGE="grafana/mimir:distroless-$IMAGE_TAG_RACE"
+          export MIMIR_ALPINE_IMAGE="grafana/mimir:$IMAGE_TAG_RACE"
           make BUILD_IN_CONTAINER=false cmd/mimir/.uptodate_race
-          docker save $MIMIR_IMAGE -o ./mimir_race_image
+          docker save $MIMIR_IMAGE -o ./mimir_race_image_distroless
+          docker save $MIMIR_ALPINE_IMAGE -o ./mimir_race_image
       - name: Upload archive with race-enabled Mimir
         uses: actions/upload-artifact@v4
         with:
           name: Race-enabled Mimir
-          path: ./mimir_race_image
+          path: |
+            ./mimir_race_image
+            ./mimir_race_image_distroless
 
   integration:
-    needs: [goversion, build]
+    needs: [goversion, build, prepare]
     runs-on: ubuntu-latest
     strategy:
       # Do not abort other groups when one fails.
       fail-fast: false
       # Split tests into 6 groups.
       matrix:
+        image_tag_prefix: ['distroless-', '']
```

Comment:
It would be nice to use `distroless` and `alpine` to be explicit.

It would also nice to get rid of `-` from `distroless-` as it looks quite ugly in the output.

Reply:
What I meant by my previous feedback about having "different box" in the UI was not to run two images via matrix, but as a competely separate job.

So we would have `integration-distroless` and `integration-alpine` jobs, running on different image. `integration-alpine` would have single precondition on entire job, and not on individual steps.

We would need to make integration job reusable first.

Reply:
I don't think github actions has a concept of job templates. Meaning to make it reusable we would have to call out into a reusable workflow (not ideal as passing artifacts would be very complex), or switch to using jsonnet to build the pipeline.

Instead I suggest we pick between using a matrix (as it currently is), or just duplicating the integration test.

Reply:
OK, let's keep the matrix for now, we can improve on this later.

Reply:
I've reworked away from the matrix to clear up some other parts.